### PR TITLE
Refactor FX resolution into its own class

### PR DESCRIPTION
**Pull request by vitek-karas**
fx_muxer is a collection of muxer functionality, framework resolution and command line parsing.
This change moves the framework resolution out into its own fx_resolver.

In the fx_resolver, made some of the methods "this" methods so that they don't have to pass the framework maps in arguments.

No functional changes, just moving code around.

### CODE REVIEW

File: src/corehost/cli/fxr/CMakeLists.txt [hunk 1]
```diff
@@ -45,6 +46,7 @@ set(HEADERS
     ./corehost_init.h
     ./fx_ver.h
     ./fx_muxer.h
+	./fx_resolver.h
```
Comment (sdmaclea) [inline src/corehost/cli/fxr/CMakeLists.txt]: indentation likely tabs -> spaces

**Comment by vitek-karas** [comment]
@sdmaclea Yes - for the FX resolver our test coverage is now pretty good. Also this change is literally just moving code between files and only one tiny change of passing two variables as member fields instead of parameters. I know the diff looks scary, not sure how else to view this so that it would show that there's really almost no difference.

File: src/corehost/cli/fxr/fx_resolver.cpp [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+
+        auto existing_framework = std::find_if(
+            fx_definitions.begin(),
+            fx_definitions.end(),
+            [&](const std::unique_ptr<fx_definition_t> & fx) { return fx_name == fx->get_name(); });
+
+        if (existing_framework == fx_definitions.end())
+        {
+            // Perform a "soft" roll-forward meaning we don't read any physical framework folders yet
+            rc = soft_roll_forward(fx_ref, false);
```
Comment (elinor-fung) [inline src/corehost/cli/fxr/fx_resolver.cpp]: Personal preference, but I like having the param name as an inline comment when it's not immediately obvious what some value (`false`) being passed is for.

Comment (elinor-fung) [reply src/corehost/cli/fxr/fx_resolver.cpp]: Or, instead maybe pull the call out of the if-else and set/pass a variable accordingly and then it should be clear from the variable name.

Comment (vitek-karas) [reply src/corehost/cli/fxr/fx_resolver.cpp]: Moving the call out of the if is not nice... added the inline comments. And also added comments on why we're setting it and what it means - as this is probably the trickiest part of the algorithm.

**Pull request merged**